src/ripple/app/consensus/RCLValidations.cpp

```diff
@@ -245,4 +245,22 @@ handleNewValidation(
     return shouldRelay;
 }
 
+std::vector<std::shared_ptr<STValidation>>
+negativeUNLFilter(
+    std::vector<std::shared_ptr<STValidation>> const& validations,
+    hash_set<NodeID> const& negUnl)
+{
+    /* Remove validations that are from validators on the negative UNL. */
```

Comment:
At your option, consider replacing with one of these two options, depending on which you find more readable:

##### Option 1: copy everything, remove what's not needed
```
std::vector<std::shared_ptr<STValidation>>
negativeUNLFilter(
    std::vector<std::shared_ptr<STValidation>> const& validations,
    hash_set<NodeID> const& negUnl)
{
    auto ret = validations;

    if (!negUnl.empty())
    {
        ret.erase(
            std::remove_if(
                ret.begin(), 
                ret.end(),
                [&negUnl](auto const& v) { return negUnl.find(v->getNodeID()) != negUnl.end(); }),
            ret.end());
    }

    return ret;
}
```

##### Option 2: copy only what's needed
```
std::vector<std::shared_ptr<STValidation>>
negativeUNLFilter(
    std::vector<std::shared_ptr<STValidation>> const& validations,
    hash_set<NodeID> const& negUnl)
{
    std::vector<std::shared_ptr<STValidation>> ret;
    ret.reserve(validations.size());

    std::copy_if(
        validations.begin(),
        validations.end(),
        std::back_inserter(ret),
        [&negUnl](auto const& v) { return negUnl.find(v->getNodeID()) != negUnl.end(); });

    return ret;
}
```

I don't think one is significantly better than the other. The first one is probably _marginally_ better for the more common case where the negative UNL is empty, but any performance improvement is likely to be below the noise floor.

Anyways, as I said, this is at your discretion; don't feel compelled to adopt either option just because I recommended it.

Reply:
Thanks. I used option 1.